packages/ckb-indexer/README.md

```diff
@@ -0,0 +1,456 @@
+```
+
+Range query for cells between given block_numbers is supported:
+
+```jsx
+cellCollector = new CellCollector(indexer, {
+  lock: {
+    code_hash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hash_type: "type",
+    args: "0xa528f2b9a51118b193178db4cf2f3db92e7df323",
+  },
+  fromBlock: "0x225510", // "0x" + 2250000n.toString(16)
+  toBlock: "0x225ce0", // "0x" + 2252000n.toString(16)
+});
+
+for await (const cell of cellCollector.collect()) {
+  console.log(cell);
+}
+```
+
+It will fetch cells between `[fromBlock, toBlock]`, which means both `fromBlock` and `toBlock` are included in query range.
+
+Page jump when query cells is supported:
+
+```jsx
+cellCollector = new CellCollector(indexer, {
+  lock: {
+    code_hash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hash_type: "type",
+    args: "0xa528f2b9a51118b193178db4cf2f3db92e7df323",
+  },
+  skip: 100,
+});
+
+for await (const tx of cellCollector.collect()) {
+  console.log(tx);
+}
+```
+
+The `skip` field represents the number of cells being skipped, which in the above code snippet means it would skip the first 100 cells and return from the 101st one.
+
+Order by block number is supported by setting `order` field explicitly:
+
+```jsx
+cellCollector = new CellCollector(indexer, {
+  lock: {
+    code_hash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hash_type: "type",
+    args: "0xa528f2b9a51118b193178db4cf2f3db92e7df323",
+  },
+  fromBlock: "0x253b40", // "0x" + 2440000n.toString(16)
+  toBlock: "0x253f28", // "0x" + 2441000n.toString(16)
+  order: "desc", // default option is "asc"
+  skip: 300,
+});
+
+for await (const cell of cellCollector.collect()) {
+  console.log(cell);
+}
+```
+
+Prefix search is supported on `args`. The default `argsLen` is -1, which means you pass the full slice of original args, and you can specify it when the `args` field is the prefix of original args.
+
+```jsx
+cellCollector = new CellCollector(indexer, {
+  lock: {
+    code_hash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hash_type: "type",
+    args: "0xa528f2b9a51118b193178db4cf2f3db92e7df3", // truncate the last byte of orignal args: 0xa528f2b9a51118b193178db4cf2f3db92e7df323
+  },
+  argsLen: 20, // default option is -1
+  fromBlock: "0x253b40", // "0x" + 2440000n.toString(16)
+  toBlock: "0x253f28", // "0x" + 2441000n.toString(16)
+  order: "desc", // default option is "asc"
+  skip: 300,
+});
+
+for await (const cell of cellCollector.collect()) {
+  console.log(cell);
+}
+```
+
+You can also set it as `any` when the argsLen of the field args might have multiple possibilities, for example, lock script's args could be 20 in normal scenario and 28 in multisig scenario, or any other length in customized scenarios. However, there's some performance lost when use `any` rather than explicitly specified length due to the low-level implementation.
```

Comment:
```suggestion
You can also set it as `any` when the argsLen has multiple possibilities. For example, lock script's args is 20 in normal scenario and 28 in multisig scenario, or any other length in customized scenarios. However, there's some performance lost when use `any` rather than explicitly specified length due to the low-level implementation.
```

Reply:
What performance lost?

Reply:
This is a copy from native indexer. after check the source code, I think there is no need those words, I'll delete it

Reply:
> What performance lost?

This is a copy from native indexer. after check the source code, I think there is no need those words, I'll delete it